quickstep/src/com/android/launcher3/appprediction/PredictionRowView.java

```diff
@@ -97,9 +102,10 @@ public void setup(FloatingHeaderView parent, FloatingHeaderRow[] rows, boolean t
     }
 
     private void updateVisibility() {
-        setVisibility(mPredictionsEnabled ? VISIBLE : GONE);
+        boolean enabled = mPredictionsEnabled && PreferenceExtensionsKt.firstBlocking(prefs2.getShowSuggestedAppsInDrawer());
+        setVisibility(enabled ? VISIBLE : GONE);
         if (mActivityContext.getAppsView() != null) {
-            if (mPredictionsEnabled) {
+            if (enabled) {
```

Comment:
`if (mPredictionsEnabled || enabled) {`

Reply:
No, `mPredictionsEnabled` is always `true` after I set Lawnchair as QS provider.

Reply:
```java
boolean enabled = mPredictionsEnabled && PreferenceExtensionsKt.firstBlocking(prefs2.getShowSuggestedAppsInDrawer());
```

What do you mean about `if (mPredictionsEnabled || enabled)`?